src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,36 @@
-import React from 'react';
+import React, { ChangeEvent } from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { filterSlice } from '../../features/filter';
+import { Status } from '../../types/Status';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const { query, status } = useAppSelector(state => state.filter);
+
+  function handleChangeQuery(newQuery: string) {
+    const normalizedQuery = newQuery.toLowerCase();
+
+    if (normalizedQuery !== query) {
+      dispatch(filterSlice.actions.changeQuery(normalizedQuery));
+    }
+  }
+
+  function handleChangeStatus(event: ChangeEvent<HTMLSelectElement>) {
+    if (status !== event.currentTarget.value) {
+      dispatch(
+        filterSlice.actions.changeStatus(event.currentTarget.value as Status),
```

Comment:
Ensure that the `Status` type includes all possible values of the `select` options ('all', 'active', 'completed'). If not, this type casting might lead to runtime errors. Consider defining `Status` as a union of these string literals if it's not already.